caikit/interfaces/runtime/data_model/info.py

```diff
@@ -39,3 +39,30 @@ class RuntimeInfoRequest(DataObjectBase):
 class RuntimeInfoResponse(DataObjectBase):
     runtime_version: Annotated[Optional[str], FieldNumber(1)]
     python_packages: Annotated[Dict[str, str], FieldNumber(2)]
+
+
+@dataobject(RUNTIME_PACKAGE)
+class ModelInfoRequest(DataObjectBase):
+    """Empty request for runtime server information"""
+
+
+@dataobject(RUNTIME_PACKAGE)
+class ModelInfo(DataObjectBase):
+    """Information regarding a specific Model instance"""
+
+    # Model information
+    model_path: Annotated[str, FieldNumber(1)]
+    name: Annotated[str, FieldNumber(2)]
+    size: Annotated[int, FieldNumber(3)]
+
+    # Module Information
+    module_id: Annotated[str, FieldNumber(4)]
+    module_name: Annotated[str, FieldNumber(5)]
+    version: Annotated[str, FieldNumber(6)]
```

Comment:
I haven't made it to the service implementation yet, but I'm assuming that it is completely agnostic to the specific model and only reports the metadata known by `caikit.runtime`. I suspect that we'll want to extend this with a freeform field that can call an overridable function on an individual `model` instance and provide model-type-specific information. Here, I would think that would look something like `metadata: Dict[str, str]`

Reply:
Sounds great. I think those other fields make sense in `metadata`